src/diffusers/models/unet_2d_condition.py

```diff
@@ -202,6 +258,7 @@ def __init__(
         conv_in_kernel: int = 3,
         conv_out_kernel: int = 3,
         projection_class_embeddings_input_dim: Optional[int] = None,
+        use_gated_attention: bool = False,
```

Comment:
```suggestion
        attention_type: str = "default",  # gated
```

When introducing new config variables let's make sure we can extend them going forward. Using a string-type variable would be nice

Reply:
> When introducing new config variables let's make sure we can extend them going forward. Using a string-type variable would be nice

@sayakpaul to fix this comment, I had to create a new repo for weights so that I can modify the unet/config file 
Weight of the model are the exact copy of original weights

Reply:
We should then submit PRs to the original model repository and tag the authors there.

Reply:
I can do that, but it will break [their fork of diffusers](https://github.com/gligen/diffusers). I'm not sure if they would prefer that

Reply:
So, IIUC the existing checkpoints from the gligen organization won't work with the current implementation that is being added in the PR?

Reply:
checkpoints are same as [original repo](https://huggingface.co/gligen/diffusers-inpainting-text-box) but NOT the config files. 

If I submit a PR to their repo to modify the config files, it will break their fork of diffusers because their fork works on a boolean `use_gated_attention` flag instead of string `attention_type` flag in this PR

Reply:
Okay, so the existing checkpoints won't work then. Could we maybe initiate a conversation with them to see if they would be willing to use the default diffusers installation instead of their fork? 

I think having the ability to use all the existing checkpoints will be crucial.

Reply:
I completely agree with you. I've reached out to their author [haotian-liu](https://github.com/haotian-liu) at liuhaotian.cn@gmail.com to checkout this PR, but haven't heard back from them

Reply:
Maybe open a discussion on their model repository? Feel free to tag me.

Reply:
Opened a discussion [here](https://huggingface.co/masterful/gligen-1-4-inpainting-text-box/commit/6d874b511f86788a5358bd82e5a2bb48763e92fb)

Reply:
Alright thanks much!

Meanwhile, I think we can knock off the other pending comments.

Reply:
Thank you. Will fix the pending comments today